recipes/master.rb

```diff
@@ -39,62 +39,27 @@ class Chef::Recipe
   end
 end
 
-directory '/etc/mesos-master/'
-
-template '/etc/default/mesos-master' do
-  source 'mesos.erb'
-  variables config: node['mesos']['master']['env']
-  notifies :run, 'bash[restart-mesos-master]', :delayed
-end
-
-# generate our flag config template but only to detect changes
-template '/etc/mesos-chef/mesos-master-config' do
-  source 'mesos.erb'
-  variables config: node['mesos']['master']['flags']
-  mode 0644
-  notifies :run, 'ruby_block[update_master_config]', :immediately
-end
-
-ruby_block 'update_master_config' do
-  block do
-    MesosHelper.update_mesos_options(node['mesos']['master']['flags'], MesosServerType::MASTER, node['mesos']['version'])
-  end
-  action :nothing
-  notifies :run, 'bash[restart-mesos-master]', :delayed
-end
-
-if node['mesos']['zookeeper_server_list'].count > 0
-  zk_server_list = node['mesos']['zookeeper_server_list']
-  zk_port = node['mesos']['zookeeper_port']
-  zk_path = node['mesos']['zookeeper_path']
-end
-
 if node['mesos']['zookeeper_exhibitor_discovery'] && node['mesos']['zookeeper_exhibitor_url']
   zk_nodes = MesosHelper.discover_zookeepers_with_retry(node['mesos']['zookeeper_exhibitor_url'])
 
   if zk_nodes.nil?
-    Chef::Application.fatal!('Failed to discover zookeepers.  Cannot continue')
+    Chef::Application.fatal!('Failed to discover zookeepers. Cannot continue.')
   end
 
-  zk_server_list = zk_nodes['servers']
-  zk_port = zk_nodes['port']
-  zk_path = node['mesos']['zookeeper_path']
-end
+  default['mesos']['master']['zk'] = 'zk://' + zk_nodes['servers'].map { |s| "#{s}:#{zk_nodes['port']}" }.join(',') + '/' +  node['mesos']['zookeeper_path']
```

Comment:
We might want to sort this list just in case the ip list ever gets reordered without actually adding or removing any ip's that really shouldn't cause a mesos master to restart.